src/services/convert.service.ts

```diff
@@ -0,0 +1,51 @@
+import { convert } from '@asyncapi/converter';
+import { AsyncAPIDocument } from '@asyncapi/parser';
+import specs from '@asyncapi/specs';
+
+import YAML from 'js-yaml';
+
+import { logger } from '../utils/logger';
+
+/**
+ * Service providing `@asyncapi/converter` functionality.
+ */
+export class ConvertService {
+  /**
+   * Convert the given spec to the desired language.
+   * @param spec AsyncAPI spec
+   * @param language Language to convert to, YAML or JSON
+   * @param version [version] AsyncAPI spec version
+   * @returns converted spec
+   */
+  public async convertSpec(
+    spec: AsyncAPIDocument | string,
+    language = '',
+    version: string = this.getLastVersion(),
+  ): Promise<string> {
+    try {
+      const asyncapiSpec = typeof spec === 'object' ? JSON.stringify(spec) : spec;
+      const convertedSpec = convert(asyncapiSpec, version);
+
+      return language === 'json'
+        ? this.convertToJSON(convertedSpec)
+        : convertedSpec;
```

Comment:
>  Of course maybe you wanna add option to convert YAML -> JSON and vice versa

yes I thought the `language` parameter was for this, so the user could send the payload as JSON, but ask for it to be converted to YAML. At least on the `studio` or `cli` I think that's how it's implemented. Should we not support this?

Reply:
Yeah, I like that option, we can leave it :) 

Of course here we have a "little" problem. When someone pass `json` spec as string (stringified JSON) then `converter-js` will return JSON object (not stringified JSON, you can check it here https://github.com/asyncapi/converter-js/blob/master/lib/index.js#L60). So we can have cases:

- someone will send `yaml` and want leave it - no problem with current implementation
- someone will send `json` and want to leave it - no problem with current implementation
- someone will send `json` and want to convert to `yaml` - we have problem because `converter-js` will return JSON and we need to transform it to the YAML. We should improve above lines and also `convertToJSON` function to support that case. I hope that you understand :)